In [14]:
#距离
import pandas as pd
import math

#缩放马氏距离
def trainScaledManhattan(in_vec):
    mean = in_vec.mean()
    std = in_vec.std()
    return mean,std


def classifyScaledManhattan(mean,std, q, prob_list):

    if(len(mean) != len(q)):
        print (len(mean))
        print(len(q))
        print("Be sure that both vectors are the same dimension!")
        return
    for i in range(len(mean)):
        if (std[i] == 0):
            std[i] = 1e-10

    return sum([prob_list[i] * abs(mean[i]-q[i])/std[i] for i in range(len(q))])

#欧氏距离
def trainEuclidean(in_vec):
    mean = in_vec.mean()
    return mean

def classifyEuclidean(mean,q, prob_list):
    if(len(mean) != len(q)):
        print (len(mean))
        print(len(q))
        print("Be sure that both vectors are the same dimension!")
        return
    return sum([prob_list[i]*math.sqrt((mean[i]-q[i])**2) for i in range(len(q))])

#根据距离搜索阀值
def search_threshold(scores,threshold):
    scores_asc = sorted(scores)
    #print(scores_asc)
    threshold_index = int(len(scores)*(1-threshold))
    return scores_asc[threshold_index]


#调整的余弦相似度量

from scipy import spatial
import numpy as np


def adjusted_cos_distance_matrix(size, matrix, row_column, prob_list):
    distances = np.zeros((size,size))
    if row_column == 0:
        M_u = matrix.mean(axis=1)
        #print('M_u is:',M_u)
        m_sub = matrix.as_matrix() - M_u[:,None]
        #print('m_sub is:',m_sub)
    if row_column == 1:
        M_u = matrix.df.as_matrix().T.mean(axis=1)
        m_sub = matrix.T - M_u[:,None]
    
    prob_Matrix = []
    for i in range(len(m_sub)):
        prob_Matrix.append(prob_list)
    print('m_sub is:',m_sub)
    
    m_sub = m_sub*np.array(prob_Matrix)
    print('secend m_sub is:',m_sub)
    
    for first in range(0,size):
        for sec in range(0,size):
            distance = spatial.distance.cosine(m_sub[first],m_sub[sec])
            distances[first,sec] = distance
            if distance < 0:
                print('distance matrix',distance)
    return distances


def adjusted_cos_distance(vector1,vector2,prob_list):
    M_u1 = vector1.mean()
    M_u2 = vector2.mean()
    m_sub1 = vector1 - M_u1
    m_sub2 = vector2 - M_u2    
    m_sub1 = np.multiply(np.array(m_sub1),np.array(prob_list))
    print(m_sub1)
    m_sub2 = np.multiply(np.array(m_sub2),np.array(prob_list))
    print(m_sub2)
    distance = spatial.distance.cosine(m_sub1,m_sub2)
    if distance < 0:
        print(distance)
    return distance

def cos_distance_matrix(size, matrix, row_column,prob_list):
    distances = np.zeros((size,size))
    if row_column == 0:
        m_sub = matrix.as_matrix() 
        #print('m_sub is:',m_sub)
    if row_column == 1:
        M_u = matrix.df.as_matrix().T.mean(axis=1)
        m_sub = matrix.T
    
    prob_Matrix = []
    for i in range(len(m_sub)):
        prob_Matrix.append(prob_list)
    
    m_sub = m_sub*np.array(prob_Matrix)
    
    
    for first in range(0,size):
        for sec in range(0,size):
            distance = spatial.distance.cosine(m_sub[first],m_sub[sec])
            distances[first,sec] = distance
    return distances

def cos_distance(vector1,vector2,prob_list):
    m_sub1 = vector1
    m_sub2 = vector2
    m_sub1 = np.multiply(np.array(m_sub1),np.array(prob_list))
    m_sub2 = np.multiply(np.array(m_sub2),np.array(prob_list))    
    distance = spatial.distance.cosine(m_sub1,m_sub2)
    if distance < 0:
        print(distance)
    return distance


#中心点对应的index,0.05false alarm的distance阀值
def center (subject,vectors,prob_list):    #根据前200个样本计算中心点
    subject_vecs = vectors[vectors['subject']==subject].drop(['subject'],axis=1)[0:200]
    distances = adjusted_cos_distance_matrix(200,subject_vecs,0,prob_list)
    #print(distances)
    distances_sum = pd.DataFrame(distances).sum().tolist()
    #print(distances_sum)
    center_index  = distances_sum.index(min(distances_sum))
    return subject_vecs.iloc[center_index]

def cosine_threshold (center_vec,vectors,prob_list):
    subject_vecs = vectors[vectors['subject']==subject].drop(['subject'],axis=1)[0:200]
    possitive_scores = []
    for i in range(len(subject_vecs)):
        distance = adjusted_cos_distance(center_vec,subject_vecs.iloc[i],prob_list)
        possitive_scores.append(distance)
    #print(sorted(possitive_scores))
    threshold = search_threshold(possitive_scores,0.05)
    return threshold
    

def test_possitive(subject,vectors,center_vec,distance_threshold,prob_list): 
    positive_vecs = vectors[vectors['subject']==subject].drop(['subject'],axis=1)[200:400]
    possitive_scores = []
    for i in range(len(positive_vecs)):
        distance = adjusted_cos_distance(center_vec,positive_vecs.iloc[i],prob_list)
        possitive_scores.append(distance)
    #print(possitive_scores)
    negative1 = sum(possitive_scores>distance_threshold)
    #print(negative1)
    return negative1

def test_passitive(subject,vectors,center_vec,distance_threshold,prob_list):
    negative_vecs = vectors[vectors['subject']!=subject].drop(['subject'],axis=1)    
    negative_scores = []
    for i in range(len(negative_vecs)):
        distance = adjusted_cos_distance(center_vec,negative_vecs.iloc[i],prob_list)
        negative_scores.append(distance)
    #print(sorted(negative_scores))
    negative2 = sum(negative_scores<distance_threshold)
    #print(negative2)
    return negative2

#from sklearn.preprocessing import StandardScaler
#ss = StandardScaler()

def train (subject,vectors,prob_list):
    subject_vecs = vectors[vectors['subject']==subject].drop(['subject'],axis=1)[0:200]  #前200个样本训练，后200个样本测试
    mean,std = trainScaledManhattan(subject_vecs)
    
    scores = []   
    for i in range(len(subject_vecs)):
        score = classifyScaledManhattan(mean,std,subject_vecs.iloc[i],prob_list)
        #print(possitive_score)
        scores.append(score)
        
    threshold = search_threshold(scores,0.05)
    #subject_vecs = ss.fit_transform(subject_vecs)    
    
    return mean,std,threshold

def test(subject,vectors,prob_list):
    mean,std,threshold = train(subject,vectors,prob_list)    #调用训练
    #print('mean:',mean)
    #print('std:',std)
    possitive_samples = vectors[vectors['subject']==subject].drop(['subject'],axis=1)[200:400]  #正样本
    #possitive_samples = ss.fit_transform(possitive_samples)
    negative_samples = vectors[vectors['subject']!=subject].drop(['subject'],axis=1)
    #negative_samples = ss.fit_transform(negative_samples)
    
    #print(possitive_samples)
    #print(negative_samples)
    
    possitive_scores = []
    negative_scores = []
    
    for i in range(len(possitive_samples)):
        possitive_score = classifyScaledManhattan(mean,std,possitive_samples.iloc[i],prob_list)
        #print(possitive_score)
        possitive_scores.append(possitive_score)
        
    for i in range(len(negative_samples)):
        negative_score = classifyScaledManhattan(mean,std,negative_samples.iloc[i],prob_list)
        #print(negative_score)
        negative_scores.append(negative_score)
    return possitive_scores,negative_scores,threshold

#纯余弦度量
#中心点对应的index,0.05false alarm的distance阀值
def center2 (subject,vectors,prob_list):    #根据前200个样本计算中心点
    subject_vecs = vectors[vectors['subject']==subject].drop(['subject'],axis=1)[0:200]
    distances = cos_distance_matrix(200,subject_vecs,0,prob_list)
    #print(distances)
    distances_sum = pd.DataFrame(distances).sum().tolist()
    #print(distances_sum)
    center_index  = distances_sum.index(min(distances_sum))
    return subject_vecs.iloc[center_index]

def cosine_threshold2 (center_vec,vectors,prob_list):
    subject_vecs = vectors[vectors['subject']==subject].drop(['subject'],axis=1)[0:200]
    possitive_scores = []
    for i in range(len(subject_vecs)):
        distance = cos_distance(center_vec,subject_vecs.iloc[i],prob_list)
        possitive_scores.append(distance)
        if distance < 0:
            print('distance threshold2:',distance)
    #print(sorted(possitive_scores))
    threshold = search_threshold(possitive_scores,0.05)
    return threshold
    

def test_possitive2(subject,vectors,center_vec,distance_threshold,prob_list): 
    positive_vecs = vectors[vectors['subject']==subject].drop(['subject'],axis=1)[200:400]
    possitive_scores = []
    for i in range(len(positive_vecs)):
        distance = cos_distance(center_vec,positive_vecs.iloc[i],prob_list)
        possitive_scores.append(distance)
        if distance < 0:
            print(distance)
    #print(possitive_scores)
    negative1 = sum(possitive_scores>distance_threshold)
    #print(negative1)
    return negative1

def test_passitive2(subject,vectors,center_vec,distance_threshold,prob_list):
    negative_vecs = vectors[vectors['subject']!=subject].drop(['subject'],axis=1)    
    negative_scores = []
    for i in range(len(negative_vecs)):
        distance = cos_distance(center_vec,negative_vecs.iloc[i],prob_list)
        negative_scores.append(distance)
        if distance < 0:
            print(distance)
    #print(sorted(negative_scores))
    negative2 = sum(negative_scores<distance_threshold)
    #print(negative2)
    return negative2

#试试欧氏距离

def train_Euclidean (subject,vectors,prob_list):
    subject_vecs = vectors[vectors['subject']==subject].drop(['subject'],axis=1)[0:200]  #前200个样本训练，后200个样本测试
    mean = trainEuclidean(subject_vecs)
    
    scores = []   
    for i in range(len(subject_vecs)):
        score = classifyEuclidean(mean,subject_vecs.iloc[i],prob_list)
        #print(score)
        scores.append(score)
        
    threshold = search_threshold(scores,0.05)
    #subject_vecs = ss.fit_transform(subject_vecs)    
    
    return mean,threshold

def test_Euclidean(subject,vectors,prob_list):
    mean,threshold = train_Euclidean(subject,vectors,prob_list)    #调用训练
    #print('mean:',mean)
    #print('std:',std)
    possitive_samples = vectors[vectors['subject']==subject].drop(['subject'],axis=1)[200:400]  #正样本
    #possitive_samples = ss.fit_transform(possitive_samples)
    negative_samples = vectors[vectors['subject']!=subject].drop(['subject'],axis=1)
    #negative_samples = ss.fit_transform(negative_samples)
    
    #print(possitive_samples)
    #print(negative_samples)
    
    possitive_scores = []
    negative_scores = []
    
    for i in range(len(possitive_samples)):
        possitive_score = classifyEuclidean(mean,possitive_samples.iloc[i],prob_list)
        #print(possitive_score)
        possitive_scores.append(possitive_score)
        
    for i in range(len(negative_samples)):
        negative_score = classifyEuclidean(mean,negative_samples.iloc[i],prob_list)
        #print(negative_score)
        negative_scores.append(negative_score)
    return possitive_scores,negative_scores,threshold

Hawkes模型提出的特征，表明激励有没有有，有的话，强度是多少。考虑有没有，我用训练集中的特征出现的概率p（发生率）的导数作为惩罚，即，如果训练集中这个特征基本没有值，如果测试集中，该特征有值，那么判断很可以能不是该用户，所以这种情况，需要对该特征产生的距离进行惩罚。惩罚系数为beta/p。beta控制惩罚的力度。
下面在计算距离时，每个特征的距离要乘以这个惩罚系数。

In [16]:
#求训练集中的各特征出现的概率
# inf_pen 是特征不出现，zero_counts = 0的情况，在 zero_counts = 1的基础上再加一个惩罚inf_pen, beta是惩罚项的影响比率
def prob(subject, vectors, samples_start, samples_end, beta, inf_pen = 10):
    subject_vecs = vectors[vectors['subject']==subject].drop(['subject'],axis=1)[samples_start:samples_end]
    prob_list = []
    for columns in [column for column in subject_vecs]:
        zero_counts = len(subject_vecs[subject_vecs[columns]==0])
        if zero_counts == 0:
            prob = samples_end - samples_start + inf_pen
        else:
            prob = (samples_end - samples_start)/zero_counts
        prob_list.append(prob**beta)
    return prob_list     

In [52]:
file_name = './data/CMU_exp_shrink_output_zero_2subject.xls'
#file_name = './data/CMU_exp_shrink_output_rep50_2subject.xls'
df = pd.read_excel(file_name)
vectors = df.drop(['key','session'],axis=1)
samples_start = 0
samples_end = 200
beta = 1
inf_pen = 1

subject_list = ['s002','s032','s036','s047','s052']
for subject in subject_list:
    prob_list  = prob(subject,vectors,samples_start,samples_end,beta,inf_pen)
    #print (prob_list)
    positive_scores,negative_scores,threshold = test_Euclidean(subject,vectors,prob_list)
    negative1 = sum(i > threshold for i in positive_scores)
    negative2 = sum(i < threshold for i in negative_scores)
    
    miss_rate1 = negative1/200
    miss_rate2 = negative2/1600
    miss_rate = (negative1+negative2)/1800
    
    
    print (subject+' miss_rate1 is:',miss_rate1)
    print (subject+' miss_rate2 is:',miss_rate2)
    print (subject+' miss_rate is:',miss_rate)

s002 miss_rate1 is: 0.105
s002 miss_rate2 is: 0.52125
s002 miss_rate is: 0.475
s032 miss_rate1 is: 0.05
s032 miss_rate2 is: 0.5375
s032 miss_rate is: 0.48333333333333334
s036 miss_rate1 is: 0.175
s036 miss_rate2 is: 0.000625
s036 miss_rate is: 0.02
s047 miss_rate1 is: 0.145
s047 miss_rate2 is: 0.33125
s047 miss_rate is: 0.31055555555555553
s052 miss_rate1 is: 0.105
s052 miss_rate2 is: 0.350625
s052 miss_rate is: 0.3233333333333333


s002 miss_rate1 is: 0.1
s002 miss_rate2 is: 0.525
s002 miss_rate is: 0.4777777777777778
s032 miss_rate1 is: 0.05
s032 miss_rate2 is: 0.535625
s032 miss_rate is: 0.4816666666666667
s036 miss_rate1 is: 0.175
s036 miss_rate2 is: 0.000625
s036 miss_rate is: 0.02
s047 miss_rate1 is: 0.135
s047 miss_rate2 is: 0.339375
s047 miss_rate is: 0.31666666666666665
s052 miss_rate1 is: 0.105
s052 miss_rate2 is: 0.34125
s052 miss_rate is: 0.315

In [81]:
import pandas as pd

file_name = './data/CMU_exp_shrink_output_zero_2subject.xls'
df = pd.read_excel(file_name)
vectors = df.drop(['key','session'],axis=1)

samples_start = 0
samples_end = 200
beta = 1
inf_pen = 1

subject_list = ['s002','s032','s036','s047','s052']
#subject_list = ['s002']
for subject in subject_list:
    prob_list  = prob(subject,vectors,samples_start,samples_end,beta,inf_pen)
    center_vec = center2(subject,vectors,prob_list)    
    distance_threshold =  cosine_threshold2(center_vec,vectors,prob_list)
    #print(distance_threshold)
    negative1 = test_possitive2(subject,vectors,center_vec,distance_threshold,prob_list)
    negative2 = test_passitive2(subject,vectors,center_vec,distance_threshold,prob_list)

    miss_rate1 = negative1/200
    miss_rate2 = negative2/1600
    miss_rate = (negative1+negative2)/1800
    
    print (subject+' miss_rate1 is:',miss_rate1)
    print (subject+' miss_rate2 is:',miss_rate2)
    print (subject+' miss_rate is:',miss_rate)

c:\users\jw\anaconda3\lib\site-packages\ipykernel_launcher.py:93: FutureWarning: Method .as_matrix will be removed in a future version. Use .values instead.


s002 miss_rate1 is: 0.085
s002 miss_rate2 is: 0.4375
s002 miss_rate is: 0.3983333333333333
s032 miss_rate1 is: 0.12
s032 miss_rate2 is: 0.725625
s032 miss_rate is: 0.6583333333333333
s036 miss_rate1 is: 0.355
s036 miss_rate2 is: 0.00375
s036 miss_rate is: 0.042777777777777776
s047 miss_rate1 is: 0.035
s047 miss_rate2 is: 0.595
s047 miss_rate is: 0.5327777777777778
s052 miss_rate1 is: 0.035
s052 miss_rate2 is: 0.2175
s052 miss_rate is: 0.19722222222222222


In [19]:
file_name = './data/CMU_exp_shrink_output_zero_2subject.xls'
#file_name = './data/CMU_exp_shrink_output_rep50_2subject.xls'
df = pd.read_excel(file_name)
vectors = df.drop(['key','session'],axis=1)

samples_start = 0
samples_end = 200
beta = 1
inf_pen = 200


subject_list = ['s002','s032','s036','s047','s052']
#subject_list = ['s002']
for subject in subject_list:
    prob_list  = prob(subject,vectors,samples_start,samples_end,beta,inf_pen)
    center_vec = center(subject,vectors,prob_list)
    distance_threshold =  cosine_threshold(center_vec,vectors,prob_list)
    print(distance_threshold)
    negative1 = test_possitive(subject,vectors,center_vec,distance_threshold,prob_list)
    negative2 = test_passitive(subject,vectors,center_vec,distance_threshold,prob_list)

    miss_rate1 = negative1/200
    miss_rate2 = negative2/1600
    miss_rate = (negative1+negative2)/1800
    
    print (subject+' miss_rate1 is:',miss_rate1)
    print (subject+' miss_rate2 is:',miss_rate2)
    print (subject+' miss_rate is:',miss_rate)

c:\users\jw\anaconda3\lib\site-packages\ipykernel_launcher.py:57: FutureWarning: Method .as_matrix will be removed in a future version. Use .values instead.


m_sub is: [[ 1.44787448 -0.75557461  0.40801282 ... -0.75557461 -0.75557461
  -0.75557461]
 [-0.7034017  -0.7034017   0.69695982 ... -0.7034017  -0.7034017
  -0.7034017 ]
 [ 2.2215668  -0.61994819  0.86150865 ... -0.61994819 -0.61994819
  -0.61994819]
 ...
 [ 3.49411711 -0.89983869  1.37198493 ... -0.89983869 -0.89983869
  -0.89983869]
 [-0.81821751 -0.81821751  1.71338113 ... -0.81821751 -0.81821751
  -0.81821751]
 [-0.74537798 -0.74537798  1.7883577  ... -0.74537798 -0.74537798
  -0.74537798]]
secend m_sub is: [[  2.54013066  -0.76708083  27.20085442 ...  -0.75937147  -0.75557461
   -0.75557461]
 [ -1.23403806  -0.7141134   46.46398828 ...  -0.70693638  -0.7034017
   -0.7034017 ]
 [  3.89748561  -0.62938903  57.43391003 ...  -0.62306351  -0.61994819
   -0.61994819]
 ...
 [  6.13003001  -0.91354182  91.4656622  ...  -0.90436049  -0.89983869
   -0.89983869]
 [ -1.43546932  -0.83067768 114.22540891 ...  -0.82232916  -0.81821751
   -0.81821751]
 [ -1.30768066  -0.75672891 119.22384685 ..

[ 5.34039743e+00 -8.91446914e-01  7.93991114e+01 -9.19450482e-01
  1.46809827e+01 -8.95995112e-01  1.44906036e+01 -8.78075210e-01
  1.50887109e+02 -8.78075210e-01  2.92037792e+02 -8.78075210e-01
  4.03880802e+02 -1.38279561e+00  3.84586539e+00 -8.86944657e-01
  3.16332397e+02 -8.78075210e-01 -8.78075210e-01  1.27017890e+03
 -1.58211750e+00 -7.21368501e+01 -2.31072424e+00 -8.82487648e-01
 -8.78075210e-01 -8.78075210e-01]
[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586
  -0.8688941  265.62717179  -0.86020515  -0.86020515 838.04883714
   1.25303858 -82.56972677  -2.26369777  -0.86452779  -0.86020515
  -0.86020515]
[-1.28164443e+00 -7.41662261e-01  8.92704669e+01 -7.64960552e-01
  1.86270213e+01 -7.45446252e-01  2.14848313e+01 -7.30537327e-01
  2.56478720e+02 -7.30537327e-01  3.61580715e+02 -7.30537327e-01
  4.35582845e+02 -1.15045248e+00  3.

  -0.86020515]
[ 7.15302857e+00 -1.05171310e+00  1.12855679e+02 -1.08475121e+00
  2.29111684e+01 -1.05707898e+00  2.92830644e+01 -1.03593741e+00
  2.64087331e+02 -1.03593741e+00  4.09501938e+02 -1.03593741e+00
  4.87333150e+02  1.90172742e+00 -2.91813354e+00 -1.04640142e+00
  3.04599856e+02 -1.03593741e+00 -1.03593741e+00  1.09668362e+03
 -1.81165739e+00 -5.52749014e+01 -2.72615107e+00 -1.04114312e+00
 -1.03593741e+00 -1.03593741e+00]
[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586
  -0.8688941  265.62717179  -0.86020515  -0.86020515 838.04883714
   1.25303858 -82.56972677  -2.26369777  -0.86452779  -0.86020515
  -0.86020515]
[   6.73796901   -1.21118678   95.39081214   -1.24923453   17.70007657
   -1.2173663    16.95897555   -1.19301898  222.60559147   -1.19301898
  347.93622866   -1.19301898  378.58751658    1.47756415   -3.29591414
   

 -6.58651134e-01 -6.58651134e-01]
[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586
  -0.8688941  265.62717179  -0.86020515  -0.86020515 838.04883714
   1.25303858 -82.56972677  -2.26369777  -0.86452779  -0.86020515
  -0.86020515]
[-1.26567806e+00 -7.32422834e-01  9.16225651e+01  1.47538216e+00
 -1.80359123e+01 -7.36159685e-01  2.02810091e+01 -7.21436491e-01
  1.50877766e+02 -7.21436491e-01  1.51452121e+02 -7.21436491e-01
  1.20981926e+02 -1.13612046e+00  9.76760772e-01 -7.28723729e-01
  8.42726882e+01 -7.21436491e-01 -7.21436491e-01 -1.63618012e+01
  9.32639942e+00 -8.68272805e+00 -7.57135359e-02 -7.25061800e-01
 -7.21436491e-01 -7.21436491e-01]
[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586

[-1.68837715e+00 -9.77030433e-01  1.40894491e+02 -1.00772249e+00
  3.20812019e+01  1.19392567e+00 -4.81187488e+01 -9.62374976e-01
  3.06652211e+02 -9.62374976e-01  4.54255492e+02 -9.62374976e-01
  4.60451980e+02 -1.51555114e+00  4.43816425e+00 -9.72095935e-01
  3.32122336e+02 -9.62374976e-01 -9.62374976e-01  1.39952829e+03
 -1.73400897e+00  1.89040236e+01  8.75580809e-02 -9.67211031e-01
 -9.62374976e-01 -9.62374976e-01]
[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586
  -0.8688941  265.62717179  -0.86020515  -0.86020515 838.04883714
   1.25303858 -82.56972677  -2.26369777  -0.86452779  -0.86020515
  -0.86020515]
[-1.45867216e+00 -8.44104703e-01  1.39043006e+02 -8.70621081e-01
  3.29016511e+01 -8.48411360e-01  4.15689976e+01 -8.31443133e-01
  3.27218235e+02 -8.31443133e-01  4.41729901e+02 -8.31443133e-01
  4.33972578e+02  1.70157046e+00 -2.

  -0.86020515]
[  5.56111041  -0.77478132  92.10389596  -0.79912     19.59419928
  -0.77873428  22.8038777   -0.7631596  208.03525655  -0.7631596
 320.67417126  -0.7631596  309.76844313  -1.20182614   2.44953538
  -0.77086828 193.7385646   -0.7631596   -0.7631596  721.62621956
  -1.37506234 -32.95242243  -2.00831473  -0.76699457  -0.7631596
  -0.7631596 ]
[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586
  -0.8688941  265.62717179  -0.86020515  -0.86020515 838.04883714
   1.25303858 -82.56972677  -2.26369777  -0.86452779  -0.86020515
  -0.86020515]
[  6.1650928   -0.75694103  95.18525932  -0.78071928  19.77287244
  -0.76080297  21.74990846  -0.74558691 145.78674511  -0.74558691
 204.86693221  -0.74558691 185.1421226   -1.17415262   1.31997548
  -0.7531181  115.69246776  -0.74558691  -0.74558691 505.24350265
   1.17676343 -12.93345921  -1.96

  -0.86020515]
[-1.54375373e+00 -8.93339725e-01  1.20452147e+02 -9.21402753e-01
  2.64403440e+01 -8.97897580e-01  3.55837964e+01 -8.79939629e-01
  4.13484912e+02 -8.79939629e-01  5.13396673e+02 -8.79939629e-01
  5.88119701e+02 -1.38573170e+00  4.36732491e+00 -8.88827908e-01
  3.52213767e+02 -8.79939629e-01 -8.79939629e-01  1.10480243e+03
 -1.58547681e+00 -4.88382805e+01 -3.04895332e-01 -8.84361436e-01
 -8.79939629e-01 -8.79939629e-01]
[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586
  -0.8688941  265.62717179  -0.86020515  -0.86020515 838.04883714
   1.25303858 -82.56972677  -2.26369777  -0.86452779  -0.86020515
  -0.86020515]
[-1.75388172e+00 -1.01493663e+00  1.43309705e+02 -1.04681946e+00
  3.43953954e+01 -1.02011488e+00  4.30581789e+01 -9.99712581e-01
  4.10238375e+02 -9.99712581e-01  5.45524932e+02  1.38449619e+00
 -3.99885032e+02 -1.5

 -8.38666130e-01 -8.38666130e-01]
[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586
  -0.8688941  265.62717179  -0.86020515  -0.86020515 838.04883714
   1.25303858 -82.56972677  -2.26369777  -0.86452779  -0.86020515
  -0.86020515]
[-1.98218092e+00 -1.14704886e+00  1.17063988e+02 -1.18308181e+00
  2.58353650e+01 -1.15290115e+00  2.87673328e+01 -1.12984312e+00
  2.08076069e+02 -1.12984312e+00  3.35838889e+02 -1.12984312e+00
  4.09601725e+02 -1.77928051e+00  3.54762297e+00 -1.14125568e+00
  2.86384712e+02 -1.12984312e+00 -1.12984312e+00  5.87334997e+02
  1.11510637e+01 -4.68865295e+01 -4.26316577e-01 -1.13552073e+00
 -1.12984312e+00 -1.12984312e+00]
[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586

[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586
  -0.8688941  265.62717179  -0.86020515  -0.86020515 838.04883714
   1.25303858 -82.56972677  -2.26369777  -0.86452779  -0.86020515
  -0.86020515]
[-1.37082344e+00 -7.93268384e-01  1.19096071e+02 -8.18187810e-01
  2.64852074e+01 -7.97315672e-01  3.64274686e+01 -7.81369359e-01
  2.17965941e+02 -7.81369359e-01  3.17164778e+02 -7.81369359e-01
  3.43634809e+02 -1.23050293e+00  3.39432703e+00 -7.89261978e-01
  3.32070534e+02 -7.81369359e-01 -7.81369359e-01  1.16014534e+03
 -1.40787272e+00  3.88590276e+01  2.27310922e-01 -7.85295838e-01
 -7.81369359e-01 -7.81369359e-01]
[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586
  -0.8688941  265.62717179  -0.86

  -0.86020515]
[-1.63013417e+00 -9.43326371e-01  1.40724898e+02 -9.72959660e-01
  3.31174554e+01 -9.48139261e-01  4.27009399e+01 -9.29176476e-01
  3.37301048e+02 -9.29176476e-01  4.98475025e+02  1.25981803e+00
 -3.71670590e+02 -1.46327004e+00  4.09902682e+00 -9.38562097e-01
  3.49160508e+02 -9.29176476e-01 -9.29176476e-01  1.17454751e+03
 -1.67419185e+00  3.18116061e+01  2.15641026e-01 -9.33845704e-01
 -9.29176476e-01 -9.29176476e-01]
[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586
  -0.8688941  265.62717179  -0.86020515  -0.86020515 838.04883714
   1.25303858 -82.56972677  -2.26369777  -0.86452779  -0.86020515
  -0.86020515]
[-1.31576892e+00 -7.61409428e-01 -4.99992191e+01 -7.85328049e-01
  3.10596727e+01 -7.65294170e-01  4.01315338e+01 -7.49988287e-01
  2.80519490e+02 -7.49988287e-01  2.75959717e+02 -7.49988287e-01
  5.03552998e+02  1.9

  -0.86020515]
[   3.11169328   -0.87025223   39.23547531   -0.89759       3.58588484
   -0.87469229   -2.83314865   -0.85719845   20.87668245   -0.85719845
   55.0712132    -0.85719845  143.36310379   -1.34991882    1.52702039
   -0.86585702  185.43136916   -0.85719845   -0.85719845  484.94870357
   12.15213944 -169.01086829   -2.25578539   -0.86150598   -0.85719845
   -0.85719845]
[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586
  -0.8688941  265.62717179  -0.86020515  -0.86020515 838.04883714
   1.25303858 -82.56972677  -2.26369777  -0.86452779  -0.86020515
  -0.86020515]
[-1.60273429e+00 -9.27470604e-01  1.35387414e+02  2.01355883e+00
 -2.28389636e+01 -9.32202597e-01  3.83085628e+01 -9.13558545e-01
  2.48388616e+02 -9.13558545e-01  2.92459084e+02 -9.13558545e-01
  5.57859239e+02 -1.43867487e+00  4.07784923e+00 -9.22786409e-01
  3.41168

  -0.86020515]
[   2.37549662   -1.00913757   22.93024393   -1.04083822   -1.9755503
   -1.01428623  -11.02379409   -0.9940005  -121.441234     -0.9940005
 -123.96717607   -0.9940005  -103.81206253   -1.56535512   -0.82061148
   -1.00404091  -52.78170329   -0.9940005    -0.9940005  -199.80160716
   27.13677656 -246.24479424   -2.6157908    -0.99899548   -0.9940005
   -0.9940005 ]
[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586
  -0.8688941  265.62717179  -0.86020515  -0.86020515 838.04883714
   1.25303858 -82.56972677  -2.26369777  -0.86452779  -0.86020515
  -0.86020515]
[-1.57491207e+00 -9.11370436e-01  4.29984445e+01 -9.39999874e-01
  5.19575226e+00 -9.16020285e-01  2.56618129e+00 -8.97699880e-01
 -2.12007462e+01 -8.97699880e-01 -4.03192564e+01 -8.97699880e-01
  1.65667487e+01 -1.41370060e+00 -9.85299774e-02 -9.06767555e-01
  2.77332091

[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586
  -0.8688941  265.62717179  -0.86020515  -0.86020515 838.04883714
   1.25303858 -82.56972677  -2.26369777  -0.86452779  -0.86020515
  -0.86020515]
[-1.46667263e+00 -8.48734413e-01  2.87880340e+01 -8.75396227e-01
  9.84285337e-01 -8.53064690e-01 -4.48787496e+00 -8.36003397e-01
 -8.36278115e+01 -8.36003397e-01 -8.35379385e+01 -8.36003397e-01
 -3.58338893e+01 -1.31654078e+00 -7.50754508e-03 -8.44447875e-01
  2.98270874e+00 -8.36003397e-01 -8.36003397e-01 -7.08772741e+01
  2.35407123e+01 -2.12543804e+02 -1.41757574e+00 -8.40204419e-01
 -8.36003397e-01 -8.36003397e-01]
[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586
  -0.8688941  265.62717179  -0.86

  -0.86020515]
[   2.35066276   -1.22074178   12.8477964    -1.25908969   -5.62440129
   -1.22697005  -19.4486335    -1.20243065 -198.25331824   -1.20243065
 -180.47698796   -1.20243065 -165.84382115   -1.89359158   -1.113314
   -1.21457642  -57.31503868   -1.20243065   -1.20243065 -201.50698129
   34.7168258  -322.18030516   -3.16429119   -1.20847302   -1.20243065
   -1.20243065]
[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586
  -0.8688941  265.62717179  -0.86020515  -0.86020515 838.04883714
   1.25303858 -82.56972677  -2.26369777  -0.86452779  -0.86020515
  -0.86020515]
[   2.24956081   -1.3222251     7.65537631   -1.36376097   -8.9598247
   -1.32897114  -25.99263807   -1.30239172 -253.36815872   -1.30239172
 -257.13418837   -1.30239172 -238.66391076   -2.05101059   -1.68529033
   -1.31554719  -97.74706686   -1.30239172   -1.30239172 -2

 -1.01308366e+00 -1.01308366e+00]
[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586
  -0.8688941  265.62717179  -0.86020515  -0.86020515 838.04883714
   1.25303858 -82.56972677  -2.26369777  -0.86452779  -0.86020515
  -0.86020515]
[ 3.28638787e+00 -1.06054791e+00  3.86669225e+01 -1.09386355e+00
  9.14452212e-01 -1.06595887e+00 -6.07184086e+00 -1.04463969e+00
 -9.02917458e+01 -1.04463969e+00 -7.49215293e+01 -1.04463969e+00
 -1.88897694e+01 -1.64510188e+00 -1.09398665e-01 -1.05519161e+00
  1.64988425e+01 -1.04463969e+00 -1.04463969e+00 -3.81112806e+01
  2.40205217e+01 -2.25176745e+02 -2.74905183e+00 -1.04988914e+00
 -1.04463969e+00 -1.04463969e+00]
[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586

  -0.86020515]
[   2.23044106   -0.70531042   22.77337557   -0.72746677    1.1548801
   -0.70890894   -4.55483724   -0.69473077  -65.46468854   -0.69473077
  -69.90076854   -0.69473077  -81.93917802   -1.0940642    -0.52483386
   -0.70174825  -35.01162182   -0.69473077   -0.69473077 -141.03114602
   17.70872543 -165.17952317   -1.82823886   -0.69822188   -0.69473077
   -0.69473077]
[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586
  -0.8688941  265.62717179  -0.86020515  -0.86020515 838.04883714
   1.25303858 -82.56972677  -2.26369777  -0.86452779  -0.86020515
  -0.86020515]
[   2.53179466   -0.87038377   25.45191893   -0.89772567    0.4328982
   -0.8748245    -6.08384923   -0.85732801  -38.53746375   -0.85732801
  -66.28399151   -0.85732801  -43.75858073   -1.35012285   -0.23306872
   -0.86598789   -0.75023689   -0.85732801   -0.85732801 -

  -0.86020515]
[ 4.52902768e+00 -6.75939949e-01  7.29330764e+01 -6.97173665e-01
  1.67745250e+01 -6.79388623e-01  1.78027483e+01 -6.65800850e-01
  1.77891524e+02 -6.65800850e-01  1.74378538e+02 -6.65800850e-01
  2.35376929e+02 -1.04850528e+00  1.81616625e+00 -6.72526111e-01
  1.93941312e+02 -6.65800850e-01 -6.65800850e-01  9.29648231e+02
 -1.19964117e+00 -4.92995089e+01 -1.75210750e+00 -6.69146583e-01
 -6.65800850e-01 -6.65800850e-01]
[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586
  -0.8688941  265.62717179  -0.86020515  -0.86020515 838.04883714
   1.25303858 -82.56972677  -2.26369777  -0.86452779  -0.86020515
  -0.86020515]
[ 4.76249714e+00 -7.49817150e-01  7.22581787e+01 -7.73371615e-01
  1.68736837e+01 -7.53642748e-01  1.81033313e+01 -7.38569893e-01
  1.89229230e+02 -7.38569893e-01  2.54539263e+02 -7.38569893e-01
  3.12080504e+02 -1.1

[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586
  -0.8688941  265.62717179  -0.86020515  -0.86020515 838.04883714
   1.25303858 -82.56972677  -2.26369777  -0.86452779  -0.86020515
  -0.86020515]
[ 5.04111182e+00 -7.84222144e-01  8.05261058e+01 -8.08857394e-01
  2.06086726e+01 -7.88223277e-01  2.58691378e+01 -7.72458811e-01
  2.36586990e+02 -7.72458811e-01  2.98630858e+02 -7.72458811e-01
  3.29223901e+02 -1.21647057e+00  2.43548743e+00 -7.80261426e-01
  2.13723197e+02 -7.72458811e-01 -7.72458811e-01  8.93705386e+02
 -1.39181768e+00 -6.59018709e+01 -2.03278635e+00 -7.76340514e-01
 -7.72458811e-01 -7.72458811e-01]
[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586
  -0.8688941  265.62717179  -0.86

[-1.16532035e+00 -6.74347819e-01  1.03910953e+02 -6.95531520e-01
  2.45095135e+01 -6.77788369e-01  3.14700673e+01 -6.64232602e-01
  2.80995047e+02 -6.64232602e-01  3.60606742e+02 -6.64232602e-01
  3.58124946e+02 -1.04603559e+00  2.80177549e+00 -6.70942022e-01
  2.31734953e+02 -6.64232602e-01 -6.64232602e-01  8.13334800e+02
 -1.19681550e+00  2.38688963e+01  1.30030827e-01 -6.67570454e-01
 -6.64232602e-01 -6.64232602e-01]
[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586
  -0.8688941  265.62717179  -0.86020515  -0.86020515 838.04883714
   1.25303858 -82.56972677  -2.26369777  -0.86452779  -0.86020515
  -0.86020515]
[-1.03507165e+00 -5.98975474e-01  7.46965412e+01 -6.17791457e-01
  1.55692762e+01 -6.02031471e-01  1.55012302e+01 -5.89990842e-01
  1.44080653e+02 -5.89990842e-01  3.88890096e+02 -5.89990842e-01
  4.49246065e+02 -9.29119436e-01  3.

  -0.86020515]
[ 4.58104420e+00 -8.05934211e-01  6.15090042e+01 -8.31251516e-01
  1.29516736e+01 -8.10046121e-01  9.75287896e+00 -7.93845198e-01
  1.54376868e+02 -7.93845198e-01  4.17396063e+02 -7.93845198e-01
  4.94193924e+02 -1.25014992e+00  3.87055639e+00  1.40304004e+00
 -1.58769040e+02 -7.93845198e-01 -7.93845198e-01  9.92165935e+02
 -1.43035171e+00 -9.90663744e+01 -2.08906631e+00 -7.97834370e-01
 -7.93845198e-01 -7.93845198e-01]
[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586
  -0.8688941  265.62717179  -0.86020515  -0.86020515 838.04883714
   1.25303858 -82.56972677  -2.26369777  -0.86452779  -0.86020515
  -0.86020515]
[ 5.42512201e+00 -9.10105259e-01  8.22694994e+01 -9.38694953e-01
  1.57934072e+01 -9.14748654e-01  1.54713147e+01 -8.96453681e-01
  2.79054197e+02 -8.96453681e-01  4.23988193e+02 -8.96453681e-01
  5.07216980e+02 -1.4

[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586
  -0.8688941  265.62717179  -0.86020515  -0.86020515 838.04883714
   1.25303858 -82.56972677  -2.26369777  -0.86452779  -0.86020515
  -0.86020515]
[  -1.26990512   -0.73486895  105.18777962   -0.75795384   20.66890346
   -0.73861828   25.9595979    -0.72384592  165.33641358   -0.72384592
  176.51867595   -0.72384592  191.15502024   -1.13991483    1.34718674
    0.50315634 -144.76918371   -0.72384592   -0.72384592   38.40270255
    9.14438157  -57.99936814   -0.32885101   -0.72748334   -0.72384592
   -0.72384592]
[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586
  -0.8688941  265.62717179  -0.86020515  -0.86020515 838.04883714
   1.25303858 -82.56

  -0.86020515]
[-1.78476695e+00 -1.03280930e+00 -6.78211440e+01 -1.06525357e+00
  2.43433841e+01 -1.03807874e+00  2.63272249e+01 -1.01731716e+00
  1.79972435e+02 -1.01731716e+00  1.99503609e+02 -1.01731716e+00
  2.69044698e+02 -1.60207427e+00  3.74873514e+00  1.35351026e+00
 -2.03463432e+02 -1.01731716e+00 -1.01731716e+00  2.20294039e+02
  1.54045827e+01 -8.53689205e+01 -5.72727073e-01 -2.20238722e-01
 -1.01731716e+00 -1.01731716e+00]
[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586
  -0.8688941  265.62717179  -0.86020515  -0.86020515 838.04883714
   1.25303858 -82.56972677  -2.26369777  -0.86452779  -0.86020515
  -0.86020515]
[-1.13323236e+00 -6.55779133e-01 -4.30628297e+01 -6.76379524e-01
  3.40409653e+01 -6.59124945e-01  4.72993685e+01 -6.45942446e-01
  3.29075239e+02 -6.45942446e-01  3.56140416e+02 -6.45942446e-01
  4.05507005e+02  1.5

[  -1.74101036   -1.01908497  -66.91991283   -1.05109811   22.21585326
   -1.02428438   23.30473101   -1.00379869  155.52523223   -1.00379869
  170.09113487   -1.00379869  208.24412364   -1.58078534    2.36563466
    0.84604544 -200.75973849   -1.00379869   -1.00379869  106.44399665
   18.42775437 -108.03424358   -0.70719465   -0.28110598   -1.00379869
   -1.00379869]
[  5.21191882  -0.87330473  78.97424377  -0.90073838  14.79540491
  -0.87776036  13.44334044  -0.86020515 200.57846485  -0.86020515
 311.99368284  -0.86020515 355.87238988  -1.35465379   2.65746586
  -0.8688941  265.62717179  -0.86020515  -0.86020515 838.04883714
   1.25303858 -82.56972677  -2.26369777  -0.86452779  -0.86020515
  -0.86020515]
[-1.46821080e+00 -8.49624525e-01 -5.57920104e+01  1.77233524e+00
 -2.09220039e+01 -8.53959344e-01  2.53689782e+01 -8.36880157e-01
  1.51004501e+02 -8.36880157e-01  1.40825036e+02 -8.36880157e-01
  1.66739409e+02 -1.31792151e+00  2.09893070e+00  7.86333319e-01
 -1.67376031e+02 -8.3688

[-1.21021404e+00 -9.24384596e-01  5.81383381e+01  2.54930970e+00
 -2.42042809e+00 -9.58086118e-01  1.43231025e+01 -9.19762673e-01
  3.72963479e+02 -9.19762673e-01  4.74819926e+02 -1.06949148e+00
  1.04415777e+01 -9.88992122e-01  2.15884760e+01 -9.19762673e-01
  7.19785013e+02 -9.33769211e-01 -9.19762673e-01  1.27209821e+03
 -1.25994887e+00 -3.90624081e+01 -4.47465375e-01 -9.43346331e-01
 -9.53121941e-01 -9.19762673e-01]
[   2.29911836   -0.58318551   22.37797815   -0.89964276    0.73255483
   -0.60444748    3.77666661   -0.58026958    8.00780289   -0.58026958
   58.30374944   -0.67473207    0.60948566   -0.62394579    0.48011664
   -0.58026958    2.73402131   -0.58910617   -0.58026958 -116.1150896
    7.51957293  -84.73433069   -2.97574145   -0.59514829   -0.60131563
   -0.58026958]
[-1.21021404e+00 -9.24384596e-01  5.81383381e+01  2.54930970e+00
 -2.42042809e+00 -9.58086118e-01  1.43231025e+01 -9.19762673e-01
  3.72963479e+02 -9.19762673e-01  4.74819926e+02 -1.06949148e+00
  1.0441577

 -8.20144493e-01 -7.91439436e-01]
[-1.21021404e+00 -9.24384596e-01  5.81383381e+01  2.54930970e+00
 -2.42042809e+00 -9.58086118e-01  1.43231025e+01 -9.19762673e-01
  3.72963479e+02 -9.19762673e-01  4.74819926e+02 -1.06949148e+00
  1.04415777e+01 -9.88992122e-01  2.15884760e+01 -9.19762673e-01
  7.19785013e+02 -9.33769211e-01 -9.19762673e-01  1.27209821e+03
 -1.25994887e+00 -3.90624081e+01 -4.47465375e-01 -9.43346331e-01
 -9.53121941e-01 -9.19762673e-01]
[-9.27624068e-01 -7.08536977e-01  5.26461033e+01  2.49175362e+00
 -1.85524814e+00 -7.34369054e-01  1.37999793e+01 -7.04994292e-01
  1.82383272e+02 -7.04994292e-01  2.18122396e+02 -8.19760805e-01
  4.24535803e+00 -7.58058379e-01  8.63632821e+00 -7.04994292e-01
  6.01801718e+02 -7.15730246e-01 -7.04994292e-01  1.20855507e+03
 -9.65745606e-01  3.03999489e+01  2.84762567e-01 -7.23071069e-01
 -7.30564033e-01 -7.04994292e-01]
[-1.21021404e+00 -9.24384596e-01  5.81383381e+01  2.54930970e+00
 -2.42042809e+00 -9.58086118e-01  1.43231025e+01 -9.1

 -9.53121941e-01 -9.19762673e-01]
[-7.95311476e-01 -6.07474092e-01  4.98231844e+01 -9.37111197e-01
  2.30780538e+00 -6.29621585e-01  1.54548697e+01 -6.04436722e-01
  1.86514053e+02 -6.04436722e-01  2.42283847e+02 -7.02833398e-01
  4.75791031e+00 -6.49931959e-01  1.12134588e+01 -6.04436722e-01
  3.29761650e+02 -6.13641342e-01 -6.04436722e-01  1.03790345e+03
 -8.27995510e-01  3.09082161e+01  4.15417549e-01 -6.19935099e-01
 -6.26359297e-01 -6.04436722e-01]
[-1.21021404e+00 -9.24384596e-01  5.81383381e+01  2.54930970e+00
 -2.42042809e+00 -9.58086118e-01  1.43231025e+01 -9.19762673e-01
  3.72963479e+02 -9.19762673e-01  4.74819926e+02 -1.06949148e+00
  1.04415777e+01 -9.88992122e-01  2.15884760e+01 -9.19762673e-01
  7.19785013e+02 -9.33769211e-01 -9.19762673e-01  1.27209821e+03
 -1.25994887e+00 -3.90624081e+01 -4.47465375e-01 -9.43346331e-01
 -9.53121941e-01 -9.19762673e-01]
[-8.98403377e-01 -6.86217655e-01  4.69128883e+01 -1.05858382e+00
  2.38230729e+00 -7.11236007e-01  1.44213490e+01 -6.8

[-1.21021404e+00 -9.24384596e-01  5.81383381e+01  2.54930970e+00
 -2.42042809e+00 -9.58086118e-01  1.43231025e+01 -9.19762673e-01
  3.72963479e+02 -9.19762673e-01  4.74819926e+02 -1.06949148e+00
  1.04415777e+01 -9.88992122e-01  2.15884760e+01 -9.19762673e-01
  7.19785013e+02 -9.33769211e-01 -9.19762673e-01  1.27209821e+03
 -1.25994887e+00 -3.90624081e+01 -4.47465375e-01 -9.43346331e-01
 -9.53121941e-01 -9.19762673e-01]
[ 2.75188637e+00 -7.91979467e-01  2.35215582e+01 -1.22173577e+00
  7.01892115e-01 -8.20853718e-01  4.20644982e-01 -7.88019570e-01
  1.69538696e+01 -7.88019570e-01  9.13221754e+01 -9.16301825e-01
  5.29332758e+00 -8.47332871e-01  1.37443014e+01 -7.88019570e-01
  4.79379208e+02 -8.00019868e-01 -7.88019570e-01  6.31099346e+02
  4.85166386e+00 -1.25461059e+02 -4.04112600e+00 -8.08225200e-01
 -8.16600590e-01 -7.88019570e-01]
[-1.21021404e+00 -9.24384596e-01  5.81383381e+01  2.54930970e+00
 -2.42042809e+00 -9.58086118e-01  1.43231025e+01 -9.19762673e-01
  3.72963479e+02 -9.19

 -9.53121941e-01 -9.19762673e-01]
[-1.00962516e+00 -7.71170976e-01  5.22819028e+01 -1.18963585e+00
  2.32096549e+00 -7.99286584e-01  1.30700527e+01 -7.67315121e-01
  2.75425779e+02 -7.67315121e-01  3.61831519e+02 -8.92226885e-01
  8.15132868e+00 -8.25070022e-01  1.93591380e+01 -7.67315121e-01
  6.02355329e+02 -7.79000123e-01 -7.67315121e-01  1.19853693e+03
 -1.05111660e+00 -5.46066840e+00 -4.05395430e-03 -7.86989868e-01
 -7.95145203e-01 -7.67315121e-01]
[-1.21021404e+00 -9.24384596e-01  5.81383381e+01  2.54930970e+00
 -2.42042809e+00 -9.58086118e-01  1.43231025e+01 -9.19762673e-01
  3.72963479e+02 -9.19762673e-01  4.74819926e+02 -1.06949148e+00
  1.04415777e+01 -9.88992122e-01  2.15884760e+01 -9.19762673e-01
  7.19785013e+02 -9.33769211e-01 -9.19762673e-01  1.27209821e+03
 -1.25994887e+00 -3.90624081e+01 -4.47465375e-01 -9.43346331e-01
 -9.53121941e-01 -9.19762673e-01]
[ 4.23835711e+00 -8.58181948e-01  4.06492826e+01 -1.32386208e+00
  1.68255095e+00 -8.89469832e-01  8.23233665e+00 -8.5

KeyboardInterrupt: 

In [13]:
a = spatial.distance.cosine([1,10000000,1,100000],[1,1,1,1])
a

-121.96928946530988